core/src/components/label/label.scss

```diff
@@ -51,6 +64,15 @@
 }
 
 
+// TODO(FW-3547): this can be removed if we
+// refactor datetime to not use an ion-item
+:host-context(.item-in-datetime) {
```

Comment:
Would it make sense to keep this all in the datetime CSS? IIRC `:host-context` isn't supported in Safari/Firefox, so this fix won't work there.

Doing the following in the datetime CSS could work:

```css
.item-in-datetime ion-label {
  width: auto;
}
```

Reply:
So I went and refreshed my memory on this because we use `:host-context` in a ton of places and there's no way we would've implemented that without it working in Safari/Firefox for those components. For example, it is used in the following components:

- [buttons](https://github.com/ionic-team/ionic-framework/blob/d669fbefd29788c5fb8fd8650279401ea311a206/core/src/components/buttons/buttons.ios.scss#L32)
- [input](https://github.com/ionic-team/ionic-framework/blob/d669fbefd29788c5fb8fd8650279401ea311a206/core/src/components/input/input.scss#L86)
- [label](https://github.com/ionic-team/ionic-framework/blob/d669fbefd29788c5fb8fd8650279401ea311a206/core/src/components/label/label.scss#L7)
- [searchbar](https://github.com/ionic-team/ionic-framework/blob/d669fbefd29788c5fb8fd8650279401ea311a206/core/src/components/searchbar/searchbar.ios.scss#L168)
- [select](https://github.com/ionic-team/ionic-framework/blob/d669fbefd29788c5fb8fd8650279401ea311a206/core/src/components/select/select.md.scss#L120)
- [textarea](https://github.com/ionic-team/ionic-framework/blob/d669fbefd29788c5fb8fd8650279401ea311a206/core/src/components/textarea/textarea.scss#L114)

In some of these cases (buttons, searchbar) it is specifically used in the `ios` mode. 

All of these are scoped components, except for select. I created a new Stencil project and made two components. One is scoped, and the other is shadow. Both have the following CSS:

```css
:host {
  color: purple;
}

:host-context(.item) {
  color: red;
}
```

When I ran the project in each browser these were the results:

![host-context](https://github.com/ionic-team/ionic-framework/assets/6577830/e50e938d-4fa0-4ebf-9c5e-f2ca6c31b16b)

So it seems that `:host-context` only works in all browsers when the component is scoped, which I think has always been the case and is the reason we use it so many places. 

In this particular case though we have full control over the item and the label so we can target it directly. I have updated the code with your recommendation: https://github.com/ionic-team/ionic-framework/pull/28146/files#diff-83857955657186aba3b158e5561a577b94c5b9b8e74aff61fc97c78f4b77343c

Reply:
Thank you!